src/VisualStudio/Core/Def/Implementation/AnalyzerDependency/AnalyzerDependencyCheckingService.cs

```diff
@@ -80,48 +80,57 @@ internal sealed class AnalyzerDependencyCheckingService
 
             var conflicts = results.Conflicts;
             var missingDependencies = results.MissingDependencies;
+            var projects = _workspace.DeferredState.ProjectTracker.ImmutableProjects;
```

Comment:
I dont know much about this code but won't this code have race if things are moved to BG?

if someone add and remove analyzers or projects before it finish running, then diagnostics will be reported even if things are already gone. (I think it already had that issue but since every ran on UI thread, it might not happened)